### ISSUE TITLE
Alternative tmux command (i.e. byobu)

### ISSUE THREAD

**Issue by kmactavish** [tony/tmuxp#143]
I didn't see any mention in the docs of being able to specify the tmux command. 

`tmuxinator` lets you specify the command with
`tmux_command: byobu`

Is tmuxp compatible with byobu (which supports the same interface as tmux)? And if so, how is it specified? If not, consider this a feature request ;).

**Comment by kmactavish** [comment]
If you run tmuxp from within a byobu session, and select `Yes' when it asks you to switch, all is well.

**Comment by tony** [comment]
It could be worth documenting if you like. If you bumped into it, I'm sure other byobu users could find it helpful. :)

**Comment by nudomarinero** [comment]
I would like to call attention to this issue again. Although it is true that manually selecting Yes within a byobu session allows to run tmuxp in byobu there are a couple of caveats:
* You have to manually interact with the system to create the byobu session and them answer to the prompt with a yes (although this could be solved using, for example, "yes | tmuxp load test.yml"). If you are using tmuxp to launch tasks in remote systems in a automated way this is problematic.
* This creates two different tmux/byobu sessions, one with the single tmuxp command and the other launched with tmuxp.
It would be nice to have the option to use directly byobu with tmuxp if possible.

**Comment by tony** [comment]
@nudomarinero @kmactavish what do you use byobu for? what does it give you over tmux/screen? 

 > You have to manually interact with the system to create the byobu session and them answer to the prompt with a yes (although this could be solved using, for example, "yes | tmuxp load test.yml"). If you are using tmuxp to launch tasks in remote systems in a automated way this is problematic.
> This creates two different tmux/byobu sessions, one with the single tmuxp command and the other launched with tmuxp.

@nudomarinero  @kmactavish Can you give me more info about how you specifically work with byobu / tmuxp?

I'm still not sure if this is an issue with byobu or tmux, or if its a bug or not

**Comment by kmactavish** [comment]
**TLDR** byobu looks nicer than default tmux. To use it, have a byobu session running somewhere before loading with tmuxp (don't have to load it from within to get most functionality). Official support looks tricky.

byobu gives you nice looking decorations with no configuration:
![nice looking decorations](http://byobu.org/img/Screenshot_from_2014-07-26_15:39:16.png)

rather than the (semi stock) tmux:
![boring](https://tmux.github.io/ss-tmux1.png)

As long as you start the tmux server with byobu before loading with tmuxp (i.e. have one byobu session running), you don't have to select yes after loading, and all you lose is some coloring (the decorations and key mappings are there).

It's definitely not a bug, just a mild incompatibility between tmuxp and byobu. I tried replacing the tmux command in tmuxp with byobu, but all of the parsing of command returns was messed up because byobu didn't strip ANSI color codes. It seemed like an endeavour to offer official support, so I was happy with the workaround.

**Comment by kmactavish** [comment]
P.S. If someone wanted to try to get it working, I would be happy. Might be an easy fix if I missed something obvious. You just change the `tmux` command to `byobu` in a couple of spots, and should start to see the weird ANSI behaviour.

**Comment by nudomarinero** [comment]
My plan is to automatically launch commands in several remote systems within byobu (or tmux if that is not possible) sessions using tmuxp. I am developing a pipeline and some steps of the commands can fail. It is very useful for debugging purposes to enter a byobu session and see what happened, etc. I prefer tmux or byobu to screen and, for me, byobu is like tmux with sugar and nice key mappings than I am used to.

I agree this is not a bug. I thought that it could be something easy to implement given that byobu is build on top of tmux. However, after the comment of @kmactavish it is clear that it may not be easy.

**Comment by akirak** [comment]
The following command allows you to initialize a byobu session using tmuxp even when there is no running byobu session:

    byobu new-session tmuxp load --yes SESSION_TO_LOAD

As you can see, this command creates a temporary byobu session that runs tmuxp and attach it immediately. The temporary session exists immediately after tmuxp loads the session. This workaround is a bit ugly on rendering but seems to always to work.

**Comment by bigcmos** [comment]
Thanks for posting this.  I was able to get my session to load in byobu by replacing `s/tmux/byobu-tmux/` in `tmux_cmd:__init__` libtmux/common.py:187.  I didn't see any side effects, other than slow load times.

about 6 seconds after replacing tmux_cmd to use byobu-tmux
```
$ time tmuxp load my_session
user	0m5.996s
```

< 1 second with regular tmux
```
$ time tmuxp load my_session
user	0m0.355s
```

< 1 second with attaching to the byobu session 
```
$ time byobu new-session tmuxp load --yes my_session
user	0m0.062s
```
